registry/api/errcode/errors.go

```diff
@@ -69,20 +69,28 @@ func (ec *ErrorCode) UnmarshalText(text []byte) error {
 // WithDetail creates a new Error struct based on the passed-in info and
 // set the Detail property appropriately
 func (ec ErrorCode) WithDetail(detail interface{}) Error {
```

Comment:
@endophage For `ErrorCode`, we can make this exception. We may want to replace it with an interface, in which case the return type will change but be consistent between `Error` and `ErrorCode`.